### PR TITLE
 Connect the FileTree component 

### PR DESCRIPTION
**Pull request by bobsilverberg**
~~Depends on #473.~~

It all works, but I'm not too sure about the tests. They feel a bit weird to me. Also, the Storybook story for `FileTree` is now broken, in that you cannot interact with it and expand the tree. I took a look but I'm not sure how to get it working.

### CODE REVIEW

**Comment by bobsilverberg** [comment]
Note that this has been rebased to reduce it down to the pertinent commit.

**Comment by bobsilverberg** [comment]
I haven't addressed the other issues yet, but I changed the code so the file tree is not rebuilt on every render. I think my approach will work, because the `FileTree` will always receive a `version` in its props.

@kumar303 Can you see if this helps the performance issue?

File: src/components/FileTree/index.tsx [hunk 1]
```diff
@@ -135,29 +137,57 @@ export type PublicProps = {
...
-type Props = PublicProps;
+type Props = PublicProps & ConnectedReduxProps;
 
 export class FileTreeBase extends React.Component<Props> {
+  tree: DirectoryNode;
+
+  constructor(props: Props) {
+    super(props);
+    const { version } = this.props;
+    this.tree = buildFileTree(
```
Comment (kumar303) [inline src/components/FileTree/index.tsx]: This optimization makes sense (good idea) but I'm not seeing much of a speedup after this change. Also, the tree should be updated when the `version` prop changes. I don't think it's worth finishing out this change for this patch. We should address the performance issues in a separate issue.

File: src/pages/Browse/index.spec.tsx [hunk 2]
```diff
@@ -179,7 +179,7 @@ describe(__filename, () => {
       versionId: String(version.id),
     });
 
-    const fileTree = root.find(FileTree);
+    const fileTree = root.find(FileTree as FileTreeBase);
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply src/pages/Browse/index.spec.tsx]: Yes, that seems better.

File: src/components/FileTree/index.tsx (hunk 1 above)
Comment (bobsilverberg) [reply src/components/FileTree/index.tsx]: Ok, I will revert the commit that introduced that change.

**Pull request merged**